trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/security/opa/OpaS3SecurityMapper.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.spi.security.opa;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.aws.proxy.spi.rest.ParsedS3Request;
+import io.trino.aws.proxy.spi.security.SecurityResponse;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.aws.proxy.spi.security.SecurityResponse.FAILURE;
+import static io.trino.aws.proxy.spi.security.SecurityResponse.SUCCESS;
+
+@SuppressWarnings("SameParameterValue")
+@FunctionalInterface
+public interface OpaS3SecurityMapper
+{
+    default Map<String, Object> toInputDocument(Map<String, Object> document)
+    {
+        /*
+            Default formats standard input:
+
+            {
+                "input": {
+                    ... nested document ...
+                }
+            }
+        */
+
+        return ImmutableMap.of("input", document);
+    }
+
+    default SecurityResponse toSecurityResponse(Map<String, Object> responseDocument)
+    {
+        /*
+            Default handles standard response:
+
+            {
+                "result": {
+                    "allow": true/false
+                }
+            }
+        */
+
+        return extractMap(responseDocument, "result")
+                .flatMap(result -> extractBoolean(result, "allow"))
```

Comment:
An interesting thing is that in OPA you can simply append `/allow` to the URL and you would get the field directly.

You would get something like:

```json
{
    "result": true
}
```

Perhaps the simplest possible implementation of this default method we could simply call `extractBoolean(responseDocument, "result")` and expect users to configure the URI appropriately?

Reply:
Do you have a pointer to the doc on this?

Reply:
There’s an example here

https://www.openpolicyagent.org/docs/latest/#4-try-opa-run-server

And docs here https://www.openpolicyagent.org/docs/latest/rest-api/#query-api

Reply:
Got it - I've made the change.